ValkeyJSON.md

```diff
@@ -0,0 +1,214 @@
+does not have a native data type for JSON. Redis Ltd.‘s RedisJSON is a popular Redis module, but not under a true 
+open source license and hence cannot be distributed freely with Valkey. There’s a demand in the Valkey 
+community to have a JSON module that matches most of the features of RedisJSON and is as API-compatible as possible. 
+See the community discussions [here](https://github.com/orgs/valkey-io/discussions?discussions_q=is%3Aopen+JSON).
+
+## Design Considerations
+
+ValkeyJSON will introduce a new JSON data type for Valkey, and commands to insert, update, delete and query JSON data. 
+To help users migrate from Redis and RedisJSON, as well as capitalize on existing OSS RedisJSON client libraries, ValkeyJSON 
+aims to be a drop-in replacement of RedisJSON. Therefore, it is designed be API-compatible and RDB-compatible with 
+Redis Ltd.’s RedisJSON.
+
+### RDB Compatibility
+
+To help users migrate from Redis and RedisJSON, ValkeyJSON will support RDB compatibility with RedisJSON. ValkeyJSON will 
+be able to load RDBs generated from RedisJSON. Likewise, RedisJSON will be able to load RDBs generated from ValkeyJSON.
+
+### Choice of JSON Library
+
+We have evaluated 7 open source JSON libraries - BSON, ION, MessagePack, yyjson, cJSON, Serde JSON, and RapidJSON. BSON, 
+ION and MessagePack do not have write (insert/update/delete) API and therefore cannot be used for the project. Among the 
+rest, RapidJSON stands out as both memory efficient and providing efficient insert/update/delete API.
+
+### JSONPath Query Parser
+
+[JSONPath](https://www.ietf.org/archive/id/draft-goessner-dispatch-jsonpath-00.html) is a query language for JSON with 
+features similar to XPath for XML. JSONPath is used for selecting and extracting elements from a JSON document. It 
+supports advanced query capabilities such as wildcard, filter expressions, slices, union, recursive search, etc.
+Using JSONPath query language, users can natively store, query, and modify JSON data structures, either wholly or partially.
+
+![](./images/jsonpath_parser_design.png)
+
+One of the deficiencies of the RapidJSON library is that it does not support JSONPath query. RapidJSON only supports 
+[JSON Pointer](https://datatracker.ietf.org/doc/html/rfc6901) query language, which is restricted to single-value selection. 
+Single-value selection corresponds to the restricted query capability in RedisJSON v1 API. RedisJSON v2 API supports a 
+much powerful JSONPath syntax with wildcard, filter expressions, slices, union, recursive search, etc., and can select 
+multiple JSON values. RedisJSON v2 API is a superset of v1 API.
+
+We should extend RapidJSON to support a subset of the JSONPath query language that is compatible with RedisJSON v2 API. 
+To achieve this, we designed a JSONPath query parser that integrates with RapidJSON. It is a recursive descent parser 
```

Comment:
From here until the end of this section as well as the diagram above appear to be design details that over-commit to a specific solution.

For example, why should it be a recursive descent parser?  For me, recursive descent implies a hand written parser: why not a generated via ANTLR or even yacc?

Reply:
Agreed it's too much design details. Reduced the description to stay at high level. BTW, I don't think recursive descent (RD) implies a hand-written parser. RD doesn't imply hand-written or code-gen. Many code-gen tools uses RD under the hood. A subset of LL(k) grammar can be implemented using RD.